src/app/custom-editor/custom-editor.component.ts

```diff
@@ -172,12 +179,48 @@ export class CustomEditorComponent implements OnInit, OnDestroy, OnChanges {
     this.setValue(value);
     this.editorContentCopy = value;
     this.rawFile = value;
-    if (value != null || value !== '') {
+    this.setContentType();
+    this.setAvailableViews();
+    if (value !== null || value !== '') {
       this.checkIfTextIsPretty();
     }
     if (this.showPrettifyOnLoad) {
-      this.onViewChange('xml');
+      if (this.contentType === 'xml') {
+        this.onViewChange('xml');
+        return;
+      }
+      if (this.contentType === 'json') {
+        this.onViewChange('json');
+        return;
+      }
+    }
+    this.onViewChange('raw');
```

Comment:
```typescript
{
  if (this.showPrettifyOnLoad && prettifiableContentTypes.includes(this.contentType)) {
    this.onViewChange(this.contentType);
    return 
  } 
  this.onViewChange('raw');
}
```

This could be nice :D